merkletree/proof.go

```diff
@@ -80,38 +80,51 @@ func (ap *AuthenticationPath) authPathHash() []byte {
 	return hash
 }
 
-func (ap *AuthenticationPath) verifyBinding(key, value []byte) bool {
+// VerifyBinding verifies both the value and the commitment
+// of the proof node of the authentication path.
+//
+// This should be called only when ap is a proof of inclusion.
+func (ap *AuthenticationPath) VerifyBinding(key, value []byte) bool {
```

Comment:
> why not just assert ap.ProofType() == ProofOfInclusion as well?

This particular check is already done at the `cc.verifyRegistration` level, isn't it? I'm referring to the explicit check of the leaf's index and the `ap.LookupIndex`. I don't see this check on the client-side anywhere anymore, and I don't think it's enough to just ensure that the server set `ap.ProofType() = ProofOfInclusion`.